sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -0,0 +1,736 @@
+    }
+
+    PagedFluxBase<SearchResult, SearchPagedResponse> search(String searchText,
+                                                            SearchOptions searchOptions,
+                                                            RequestOptions requestOptions,
+                                                            Context context) {
+        SearchRequest searchRequest = this.createSearchRequest(searchText, searchOptions);
+        return new PagedFluxBase<>(
+            () -> this.searchFirstPage(searchRequest, requestOptions, context),
+            nextPageParameters -> this.searchNextPage(searchRequest, requestOptions, nextPageParameters, context));
+    }
+
+    /**
+     * Retrieves a document from the Azure Cognitive Search index.
+     *
+     * @param key the name of the document
+     * @return the document object
+     */
+    public Mono<Document> getDocument(String key) {
+        return this.getDocumentWithResponse(key, null, null)
+            .map(Response::getValue);
+
+    }
+
+    /**
+     * Retrieves a document from the Azure Cognitive Search index.
+     *
+     * @param key document key
+     * @param selectedFields selected fields to return
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @return the document object
+     */
+    public Mono<Document> getDocument(
+        String key,
+        List<String> selectedFields,
+        RequestOptions requestOptions) {
+        return this.getDocumentWithResponse(key, selectedFields, requestOptions)
+            .map(Response::getValue);
+    }
+
+    /**
+     * Retrieves a document from the Azure Cognitive Search index.
+     *
+     * @param key document key
+     * @param selectedFields selected fields to return
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @return a response containing the document object
+     */
+    public Mono<Response<Document>> getDocumentWithResponse(
+        String key,
+        List<String> selectedFields,
+        RequestOptions requestOptions) {
+        return withContext(context -> getDocumentWithResponse(key, selectedFields, requestOptions, context));
+    }
+
+    Mono<Response<Document>> getDocumentWithResponse(
+        String key,
+        List<String> selectedFields,
+        RequestOptions requestOptions,
+        Context context) {
+        return restClient
+            .documents()
+            .getWithRestResponseAsync(key, selectedFields, requestOptions, context)
+            .map(res -> {
+                Document doc = res.getValue();
+                DocumentResponseConversions.cleanupDocument(doc);
+                return new SimpleResponse<>(res, doc);
+            })
+            .onErrorMap(DocumentResponseConversions::exceptionMapper)
+            .map(Function.identity());
```

Comment:
Same question about this `map`.

Reply:
since getWithRestResponseAsync returns a Mono of SimpleResponse (autogenerated code). The map(Function.identity()) allows returning a Mono of Response